### PR TITLE
Docs/update release notes

### PR DESCRIPTION
**Pull request by MiguelMarcelino**
I added some extra notes to the release notes that hopefully clarify the release process. Feel free to comment if anything is not according to the remaining docs or if you do not agree. 

I updated some links from handbook to Slite, as these were migrated already. Feel free to comment in case I picked the wrong documents or if there are newer ones we should be using.

### CODE REVIEW

File: RELEASE.md
Comment (MiguelMarcelino) [reply RELEASE.md:224]: I see tons of AI-suggested solutions here. Should we accept them?

Comment (nicklem) [reply RELEASE.md:224]: If I understand correctly, the fixes are actually suggested by `markdownlint` (we can do this only for JS, TS, and markdown so far, see [this table](https://docs.codacy.com/getting-started/supported-languages-and-tools/#supported-languages-and-tools)), with the instructions enriched by AI.

You can ignore these suggestions; the tool checks all edited lines and the file is very likely full of such minor issues.

File: RELEASE.md
Comment (nicklem) [inline RELEASE.md:190]: We can simplify this step since the instructions you mention are described in the release notes README you link to. It's best to completely defer to that file and have only one source of truth.

```suggestion
    Make sure that you meet all the requirements specified in the [release notes README file](https://github.com/codacy/release-notes-tools/blob/master/README.md#requirements) and then run the makefile target `get_release_notes` to automatically generate the release notes for the new version:
```

File: RELEASE.md
Comment (nicklem) [inline RELEASE.md:258]: Tweak to match the official name of the CLI

```suggestion
-   [ ] 2.  Tag the [Codacy Analysis CLI tool](https://github.com/codacy/codacy-analysis-cli) and [CLI Coverage Reporter](https://github.com/codacy/codacy-coverage-reporter) with the version of the release being done.
```

File: RELEASE.md
Comment (nicklem) [inline RELEASE.md:270]: Tweak for conciseness

```suggestion
    3.  For the CLI Coverage Reporter, you must update the `SELF_HOSTED_CODACY_REPORTER_VERSION` flag on the `get.sh` script found in the project's root. This should match the version used by the new self-hosted release. You do not have to do this for the Analysis CLI tool.
```

**Comment by nicklem** [comment]
Thanks @MiguelMarcelino! I left some comments with tweaks, mostly to polish the clarity of the instructions. Looking pretty good already!

I suggest waiting for @prcr to return and take a look since he has lots of experience with the process.

File: RELEASE.md
Comment (prcr) [reply RELEASE.md:224]: The issue is that the markdownlint rules for this repository aren't defined properly (for instance, someone may have applied a test coding standard to repository or something like that and the rules are now broken). Let's just ignore them at this point. :+1:

File: RELEASE.md
Comment (prcr) [inline RELEASE.md:58]: It's great that you're giving explicit visibility to all these validations and checks @MiguelMarcelino! :muscle: Typically, these go a bit under the radar and as you mention it can cause trouble further down the line.

However, I think these validations should even be performed at an earlier stage, more specifically this is what should be validated when we mention "getting a status from each team" and "confirming if the release can go ahead" on [lines #L37-L39 above](https://github.com/codacy/chart/pull/777/files#diff-2b1b69303b927a484e02c7fad9fc87d0d3ff0dc22ae1da0ecd0dc935d922a23cR37-R39).

File: RELEASE.md
Comment (prcr) [reply RELEASE.md:190]: I agree with @nicklem, let's avoid duplicating the requirements here so everything is always centralized and up to date on the codacy/release-notes-tools README.

Also, just to explain why this step didn't provide much more context before, my original idea was that the output of running the Makefile target should guide the Release Manager through fixing any setup issues and point them to the requirements on the README. But I do agree that it makes more sense to leave things explicit here. :+1:

File: RELEASE.md
Comment (prcr) [inline RELEASE.md:224]: Here we can link directly to the section about exploratory tests:

```suggestion
-   [ ] 1.  Do [exploratory tests](https://codacy.slite.com/app/docs/xonSwt3M2qds8m#a649c63d) around the functionality the new features impacted to make sure everything is running as it should, and raise bugs or concerns if any.
```

File: RELEASE.md
Comment (prcr) [reply RELEASE.md:258]: Yes, but even better would be to use the "official" names of both the Codacy Analysis CLI and Codacy Coverage Reporter. :nerd_face: 

```suggestion
-   [ ] 2.  Tag the [Codacy Analysis CLI](https://github.com/codacy/codacy-analysis-cli) and [Codacy Coverage Reporter](https://github.com/codacy/codacy-coverage-reporter) CLI tools with the version of the release being done.
```

File: RELEASE.md
Comment (prcr) [reply RELEASE.md:270]: As above, let's use the "official" names:

```suggestion
    3.  For the Codacy Coverage Reporter, you must update the `SELF_HOSTED_CODACY_REPORTER_VERSION` flag on the `get.sh` script found in the project's root. This should reflect the new version that is intended to be used in the new self-hosted release. You do not have to do this for the Codacy Analysis CLI.
```

**Review by prcr — APPROVED** [review]
These are amazing improvements to the Self-hosted release process @MiguelMarcelino! :raised_hands: I'm already approving because despite @nicklem's and my own suggestions, this new version already includes important details that were missing before.

If you decide to apply some of our feedback and need another approval just let me know.

Also, I'm afraid I took quite some time to realize that this pull request was waiting for my feedback. :sweat:

**Comment by MiguelMarcelino** [comment]
This was just sitting here. Will close for now and reopen if needed.

**Pull request closed without merging**
